packages/internal-test-env/index.ts

```diff
@@ -78,100 +77,97 @@ export interface TestingEnvironmentBase {
 type FeatureFlags = {
   [key: string]: any;
 };
+const ENV_VAR_PREFIX = "E2E_TEST_FEATURE_";
+
 let envLoaded = false;
-let featuredFlags: FeatureFlags = {};
 export interface EnvVariables {
   // Common Envs
-  E2E_TEST_ENVIRONMENT: AvailableEnvironment;
+  E2E_TEST_ENVIRONMENT: AvailableEnvironments;
   E2E_TEST_IDP: string;
 }
 
 export function setupEnv() {
-  // If we're in CI, the environment is already configured.
-  if (process.env.CI) {
-    return;
-  }
-
-  if (envLoaded) {
+  // If we're in CI, the environment is already configured. Otherwise, it is
+  // loaded once.
+  if (process.env.CI || envLoaded) {
     return;
   }
-
   const envPath = join(process.cwd(), "e2e/env/.env.local");
 
-  // Otherwise load dotenv configuration
+  // Load dotenv configuration
   config({
     path: envPath,
   });
 
   if (!process.env.E2E_TEST_ENVIRONMENT) {
     console.error(
-      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
-tried looking in the following directory for \`.env.local \`: 
-${envPath}`
+      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT, tried looking in the following directory for '.env.local': ${envPath}`
     );
   }
-
-  // Creating feature flag object if there are feature flags
-  Object.keys(process.env)
-    .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
-    .forEach((key) => (featuredFlags[key] = process.env[key]));
-  // Marking env as loaded
   envLoaded = true;
 }
 
-function getTestingEnvironment(
-  environment: unknown
-): asserts environment is EnvVariables {
-  // Populate your process.env from your .env file of choice
+function getBaseTestingEnvironment<T extends LibraryVariables>(
+  libVars?: T
+): T extends NodeVariables
+  ? TestingEnvironmentNode
+  : TestingEnvironmentBrowser {
+  setupEnv();
+
+  // Load and validate target environment name.
+  const targetEnvName = process.env.E2E_TEST_ENVIRONMENT;
+  if (!availableEnvironments.includes(targetEnvName as AvailableEnvironments)) {
+    throw new Error(`Unknown environment: [${targetEnvName}]`);
+  }
 
-  // TODO: Replace these inline validations and checks with envalid or env-var
-  if (
-    !availableEnvironment.includes(
-      (environment as EnvVariables).E2E_TEST_ENVIRONMENT as AvailableEnvironment
-    )
-  ) {
+  // Load and validate target OpenID Provider.
+  const targetOp = process.env.E2E_TEST_IDP;
+  if (typeof targetOp !== "string" || !isValidUrl(targetOp)) {
```

Comment:
```suggestion
  const targetIdp = process.env.E2E_TEST_IDP;
  if (typeof targetIdp !== "string" || !isValidUrl(targetIdp)) {
```

I think that might be clearer?

Reply:
OP is a standard OpenID terminology (for OpenID Provider), and I'd prefer we use spec terminology where we can. Changing the env var name has repercussions across all of our repos, so it's not worth the effort though.

Reply:
Okay, that's fine then, but let's add support for `E2E_TEST_OP` (though I've only ever heard people inside the company talk of this as IdP), and then we can do a PR to e2e-environments and roll out `E2E_TEST_OP` to all environments and then when we upgrade the developer tools in each SDK, we can switch the CI pipelines across to the new environment variable.

But if we're saying we don't want to migrate to `E2E_TEST_OP`, then I'd rather that this code use `targetIdp` and not `targetOp` for clarity & consistency.

Reply:
OP is what the OpenID spec uses, and how everyone in the auth community calls it. We should align on calling it so everywhere, but I'll create a ticket to do it separately not to expand the scope of this specific PR.